darts/models/forecasting/regression_ensemble_model.py

```diff
@@ -144,17 +166,18 @@ def fit(
                 self.train_n_points, series
             )
 
-        for model in self.models:
-            # maximize covariate usage
-            model._fit_wrapper(
-                series=forecast_training,
-                past_covariates=past_covariates
-                if model.supports_past_covariates
-                else None,
-                future_covariates=future_covariates
-                if model.supports_future_covariates
-                else None,
-            )
+        if self.retrain_forecasting_models:
```

Comment:
It seems a bit strange to me that we ues `model.fit_wrapper()` here but [below](https://github.com/unit8co/darts/blob/8e93720e6277726089058cf4d039b97de0f57c34/darts/models/forecasting/regression_ensemble_model.py#L178) (before end of this method) we call `model.fit()` directly. Do you know if this was intended? If not intended then it's probably be better to use the same logic.

As far as I see it, _fit_wrapper() is handling the passing of covariates to the model based on their support.
We could probably refactor this method with the logic from [below](https://github.com/unit8co/darts/blob/8e93720e6277726089058cf4d039b97de0f57c34/darts/models/forecasting/regression_ensemble_model.py#L178). Then we can also avoid overwriting by the subclasses.

Reply:
Switched to fit_wrapper in all relevant places!